telemetry/firmware/telemetry.c

```diff
@@ -0,0 +1,316 @@
+/*
+   Header:
+    This is the functional code for the telemetry board
+    Function: Responsible for pulling safety & failure critical information from CANBus and passing it along to UART for telemetry RF module
+            : Interface with RF Module 
+            : Interface with CANBus
+
+
+    confirm UART setup
+    confirm CAN setup
+    setup UART sending  
+
+   Author:
+    adi ramachandran
+    aramachandran@olin.edu
+    6307969467
+
+General notes: 
+- Both sides need LUT with: UART_id: {data len (bytes), send frequency (Hz)} 
+- UART Data sends acheived with set of queues 
+- 
+
+General questions: 
+- need a state machine for CAN inputs? 
+
+
+CAN quesitons: 
+    how to implement 28 CAN addr reads successfully? 
+
+        mobnb0 with can mailboxes
+        can multiple can messages share the same mailbox? do we prioritize the msg priority when figuring out which mailboxes to use? 
+
+        msg's we care about lol 
+        0xB
+        0xC
+        0xE
+        0x40-0x57
+        0x58-0x63
+        0xc0
+        0x10
+
+        do i create a custom mask that fits my needs? 
```

Comment:
The range of IDs goes from 0xB to 0x63, so we can just have a mask that listens to any message in that range. So 0x7F should work as a mask.

Reply:
Actually on second thought, we should just implement a MOb for 0xB, 0xC, 0xE, and 0x10 with the mask 0x1F, and then have a MOb for 0xc0 that only listens for that, and a MOb for 0x40-0x64.

Reply:
I got the mask part wrong.

In a mask, if a bit is 0, that bit is ignored, and if the bit is 1, the bit must patch. So if you want to accept messages from 0x40 to 0x64, your mask will be 0x7C0.

Reply:
But it will also accept messages up to 0x7F, so you'll need to do some additional checking